flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -48,8 +48,9 @@ public static String getHostnameFromFQDN(String fqdn) {
 	 * Works also for ipv6.
 	 *
 	 * See: http://stackoverflow.com/questions/2345063/java-common-way-to-validate-and-convert-hostport-to-inetsocketaddress
+	 * @return URL object for accessing host and Port
 	 */
-	public static void ensureCorrectHostnamePort(String hostPort) {
+	public static URL ensureCorrectHostnamePort(String hostPort) {
```

Comment:
Since it is now returning something, could we change the name to reflect it properly, like ``generateCorrectUrlFromHostnameAndPort``